### ISSUE TITLE
Fix S3240: FP when using having lambdas within the IF-ELSE

### ISSUE THREAD

**Issue by eduherminio** [SonarSource/sonar-dotnet#2244]
### Description

False positive in [S3240](https://jira.sonarsource.com/browse/RSPEC-3240) when using `Task.Run(Action)`

### Repro steps

Please provide the steps required to reproduce the problem

```C#
if(condition())
{
    Task.Run(() => A();
}
else
{
    Task.Run(() => B();
}
```

### Expected behavior

Nothing.

### Actual behavior

Suggested refactoring:
```C#
Task.Run(
    condition()
    ? () => A()
    : () => B());
```

Which leads to:
`CS0173	Type of conditional expression cannot be determined because there is no implicit conversion between 'lambda expression' and 'lambda expression`

### Related information

* **SonarC# Version** 7.8 (build 7320)
* **Visual Studio Version** VS 2017 15.9.2
